vulnfeeds/git/repository.go

```diff
@@ -26,6 +27,11 @@ func (v Versions) Swap(i, j int)      { v[i], v[j] = v[j], v[i] }
 // RepoTagsMap acts as a cache for RepoTags results, keyed on the repo's URL.
 type RepoTagsMap map[string]Versions
 
+// NormalizedRepoTagsMap allows for looking up a repo's tags by normalized values.
+// A composite struct key of repo and normalized tag is not used to allow for
+// fuzzy matches on all available normalized versions for a given repo
+type NormalizedRepoTagsMap map[string]map[string]Version
```

Comment:
nit: this is a little hard to understand. We should either document the expected keys/nesting structure. i.e.

```
repo URL -> normalized verison -> original unnormalized version
```

Or restructure some of these data structures to have some better abstraction. Something like: 

```go
// NormalizedVersions provides mappings from normalized versions to their original tag value.
type NormalizedVersions struct {
  OriginalVersion map[string]string
}

// RepoNormalizedVersionsCache is a cache allowing for lookups of normalized versions per repo. 
type RepoNormalizedVersionsCache struct {
  Repo map[string]NormalizedVersions
}

// Usage
normalizedVersionsCache.Repo[url].OriginalVersion[v]
```

Reply:
Thanks, I've given this a thorough overhaul (which conveniently surfaced some bugs in both my understanding of maps in Go and my caching logic). PTAL.